src/core/agents.jl

```diff
@@ -232,8 +235,11 @@ macro agent(new_name, base_type, super_type, extra_fields)
             end
             # @show expr # uncomment this to see that the final expression looks as desired
             # It is important to evaluate the macro in the module that it was called at
-            Core.eval($(__module__), expr)
+            Core.eval($(__module__), Core.@__doc__(expr))
```

Comment:
This extra `@__doc__` probably wasn't meant to be here? Probably won't end up doing the right thing.

Reply:
Whoops, thanks for catching that. That was left over from earlier. I think it's time to call it a night, I'm getting tired :sweat_smile: